qfieldsync/gui/layers_config_widget.py

```diff
@@ -128,7 +131,26 @@ def reloadProject(self):
 
         self.layersTable.setRowCount(0)
         self.layersTable.setSortingEnabled(False)
+
+        # Get filtered layers
+        show_visible_only = self.showVisibleLayersOnlyCheckbox.isChecked()
+        filter_text = self.textFilterBox.text().lower()
+
         for layer_source in self.layer_sources:
+            layer_name = layer_source.layer.name().lower()
+            layer_visible = (
+                QgsProject.instance()
+                .layerTreeRoot()
+                .findLayer(layer_source.layer.id())
+                .isVisible()
+            )
```

Comment:
@SeqLaz, just a minor suggestion: you could avoid the `layer_visible` always being evaluated on all layers and instead evaluate it only if `show_visible_only` is active.

Reply:
Great idea, I already push a commit!